### ISSUE TITLE
Support for verifyEd25519Signature

### ISSUE THREAD

**Issue by mateusap1** [Hyperion-BT/Helios#28]
A common functionality, currently supported by Plutus-Tx, is to validate signatures inside the script logic.

## Use Case
This functionality can be useful for many applications, but specially oracles. As one example, a DEX that want's to get the USD price and contains a list of trusted public key hashes (the oracles) could benefit from this feature. Instead of requiring those oracles to update scripts themselves and change their datum to the current USD price, they could simply sign a message with the right value and the script would validate that the majority (or the consensus) provided a valid signature and the same value.

## Proposal
I propose that we implement the `verifyEd25519Signature` function, allowing public keys to be used to validate signatures. Implementation details can be discussed on this comment section.

**Comment by christianschmitz** [comment]
We could implement another type-safe wrapper for `ByteArray` called `PubKey`.

This could help to avoid confusion with `PubKeyHash` (blake224 hash of `PubKey`).

`verify` could be implemented as a method of `PubKey`:
```
pub_key.verify(message: ByteArray, signature: ByteArray) -> Bool
```

I don't think a type-safe wrapper for the signature is necessary.

Constructing a new `PubKey` could look as follows:
```
PubKey::new(bytes: ByteArray) -> PubKey
```

**Comment by christianschmitz** [comment]
Note: there is no on-chain way of going from `PubKey` to `PubKeyHash` (only blake256 is available)

**Comment by christianschmitz** [comment]
Update: has been implemented on `dev` branch

**Comment by christianschmitz** [comment]
merged into main as of v0.9.3
